src/civetweb.c

```diff
@@ -17060,6 +17090,26 @@ mg_start(const struct mg_callbacks *callbacks,
 		return NULL;
 	}
 
+	/* listen_ctrl_sd is used by mg_stop to make master thread
+	   break out safely from i/o wait in poll instead of relying on 
+	   frequent poll timeout to wake up and check the stop flag.
+	*/
+	ctx->listen_ctrl_sd = socket(AF_INET, SOCK_DGRAM, 0);
```

Comment:
we need to handle listen_ctrl_sd error case by returning NULL like rest of the error in mg_start